Backend/DataAccessLayer/Repositories/IResetRepository.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repositories
+{
+    interface IResetRepository
+    {
+        void AddResetID(string email, string resetID, DateTime expirationTime);
+        void DeleteResetID(string resetID);
+        DateTime GetExpirationTime(string resetID);
+        Guid GetUserID(string resetID);
+        string GetResetID(Guid userID);
+        string GetResetID(string email);
+        bool ExistingResetIDGivenResetID(string resetID);
+        bool ExistingResetIDGivenEmail(string email);
+        bool UpdatePassword(Guid userIDToChangePassword, string newPasswordHash);
+        List<string> GetSecurityQuestions(Guid userIDToGetQuestionsFrom);
+        bool CheckSecurityAnswers(Guid userIDToGetQuestionsFrom, List<string> userSubmittedSecurityAnswers);
+        void LockOut(string resetID);
+        bool CheckLockOut(string resetID);
+        int GetAttemptsPerResetID(string resetID);
+        int GetResetIDCountPerEmail(string email);
+    }
```

Comment:
Having a general method to return the PasswordReset Object by passing passwordReset ID would save the need for a large amount of the functions as you could retrieve associated variables from the object instead of specific variables.

Add method: PasswordReset getPasswordReset(DatabaseContext _db, GUID passwordResetID)

Only methods really needed are basic crud operations:
PasswordReset createPasswordReset((DatabaseContext _db, PasswordReset newPasswordReset)
PasswordReset getPasswordReset((DatabaseContext _db, GUID passwordResetID)
PasswordReset  deletePasswordReset((DatabaseContext _db, GUID passwordResetID)
PasswordReset updatePasswordReset((DatabaseContext _db, GUID passwordResetID)